app/models/miq_worker.rb

```diff
@@ -329,15 +329,23 @@ def send_message_to_worker_monitor(message, *args)
   end
 
   def start
-    msg = "Worker started: ID [#{self.id}], PID [#{self.pid}], GUID [#{self.guid}]"
-    MiqEvent.raise_evm_event_queue(self.miq_server, "evm_worker_start", :event_details => msg, :type => self.class.name)
+    pid = fork do
+      ActiveRecord::Base.connection.reconnect!
+      DRb.stop_service
+      require "workers/#{self.class.corresponding_runner.underscore}"
+      self.class.corresponding_runner.constantize.start_worker(command_line_params)
+      exit!
```

Comment:
@gtanzillo saw this... I don't think this line will ever be reached since each worker runs and only exits when they get the interrupts or the server process tells them to exit.